lib/chef/provisioning/docker_driver/driver.rb

```diff
@@ -105,7 +110,8 @@ def build_container(machine_spec, machine_options)
         target_repository = 'chef'
```

Comment:
I agree with @mwrock - it should be `machine_spec.from_image`.  That supports the following:

```ruby
machine "1" do
  from_image "my_image"
 ...
end

machine_image "2" do
  from_image "my_image"
  ...
end```

We want to support that rather than
```ruby
machine_image "2" do
  machine_options :docker_options => 
    :from_image => ...
  end
end
```